[cuenimby] Add Enhanced CueNimby System Tray with OpenCue Icons, Emoji Status, and Connectivity Awareness

KernAttila opened the pull request:
**Link the Issue(s) this Pull Request is related to.**
- https://github.com/AcademySoftwareFoundation/OpenCue/issues/2069
- #1800
- https://github.com/AcademySoftwareFoundation/OpenCue/issues/2054

**Summarize your change.**

This PR modernizes CueNIMBY with a Qt6-based system tray, professional icons, enhanced status detection, and comprehensive documentation.

**CueNIMBY Application Changes:**
- Added professional icons featuring the OpenCue logo for all status states
- Migrated backend from pystray to Qt6 QSystemTrayIcon for native cross-platform support
- Added emoji status hints (🔒❌⚠️🔧) in notifications and tooltips for quick visual recognition
- Implemented resilient connection: starts even when CueBot is unreachable and reconnects automatically
- Enhanced status detection with distinct states:
    - STARTING, AVAILABLE, WORKING, NIMBY_LOCKED, HOST_LOCKED, HOST_DOWN
    - CUEBOT_UNREACHABLE (new), NO_HOST (new), HOST_LAGGING (new), REPAIR (new), UNKNOWN
- Added intelligent menu states: options appear disabled when actions cannot be performed
- New "Launch CueGUI" menu item for quick access to CueGUI (disabled when not available)
- New "Open Config File" menu item to edit configuration directly from tray
- Enhanced "About" dialog showing CueBot address, monitored hostname, and version
- Improved notifications with helpful troubleshooting information about CueBot and RQD status
- Moved config file path to `~/.config/opencue/cuenimby.json` to align with other OpenCue modules
- Added CueNIMBY to CI test pipeline

**Documentation Changes:**
- Updated all CueNIMBY documentation with Qt6 architecture and new features
- Added platform-specific screenshots (macOS and Windows) to 3 key documentation files:
    - `quick-start-cuenimby.md`: Setup and common issues
    - `cuenimby-tutorial.md`: Step-by-step configuration with visuals
    - `cuenimby-user-guide.md`: Menu features and troubleshooting
- Created visual icon gallery showing all professional OpenCue icons across documentation
- Added new developer guide: `hybrid-rqd-setup.md` with complete instructions for:
    - Running CueBot in Docker alongside native RQD instances
    - Platform-specific setup for macOS, Windows, and Linux
    - Port configuration to avoid Docker/native RQD conflicts
    - Testing CueNIMBY in realistic hybrid environments
- Updated `cuenimby/README.md` with comprehensive feature list and icon reference
- Updated NIMBY concepts documentation with enhanced status explanations
- All documentation uses Jekyll absolute paths for proper rendering

**Configuration Changes:**
- Added Linux RQD configuration example: `sandbox/rqd_linux.conf`
- Updated hybrid RQD setup guide with Linux support alongside macOS and Windows
- Configuration files now include NIMBY settings optimized for workstation testing

**Assets:**
- Added 9 professional PNG icons with OpenCue logo to `cuenimby/cuenimby/icons/`
- Added 17 macOS screenshots showing all status states and menu options
- Added 10 Windows screenshots for cross-platform documentation coverage

Co-authored-by: Kern Attila GERMAIN <5556461+KernAttila@users.noreply.github.com>
Co-authored-by: Ramon Figueiredo <rfigueiredo@imageworks.com>

KernAttila commented:
@ramonfigueiredo I added icons with the OpenCue logo. I'm currently testing on Windows, then I'll make this PR reviewable so you can test it :)

ramonfigueiredo commented:
> @ramonfigueiredo I added icons with the OpenCue logo. I'm currently testing on Windows, then I'll make this PR reviewable so you can test it :)

Great. Thanks, @KernAttila

ramonfigueiredo commented:
Great work, @KernAttila . Please incorporate Jimmy’s comments and update the CueNIMBY documentation to reflect the latest changes. Thank you for your valuable contribution.

ramonfigueiredo commented:
Thanks for sharing the screenshot, @KernAttila . That's great!

It is on my to-do list to add the screenshot to the documentation as well.

@lithorus 
Feel free to post the screenshots of the Linux here as well. That will help you when I update the documentation. 

Thanks!

lithorus commented:
> @lithorus, @ramonfigueiredo The last 3 commits are the QtPy version. It works well on Windows :) I added the REPAIR mode+icon and made the menu items update according to the current status (ex: disabling the "Set Available/Disabled" menu when the switch is not possible) Let me know :) I'm also thinking about additional menus from the Config file, would be handy to have a "Start RQD" option with customizable command.

Good idea about more items, however I think start cuegui would be more usefull. In some deployments RQD is either run as root or as a seperate `render` user.

KernAttila commented:
> Looking really nice. There is just the disable/enable notification that seems a bit weird.
> <img alt="image" src="https://github.com/user-attachments/assets/b6e0c794-435c-43e7-8378-96372ce2ef85" />
> Notifications are enabled, but it says disabled (you click it to disable it)  with a check box. It's a bit confusing.
> 
> There is also the config file is inside the `~/.opencue` folder on Linux. It should be in `~/.config/opencue` like eg. `cuesubmit.conf`

Ah yep that's confusing, I'll rollback the labels to keep the checkboxes and move the config file.
Thanks for the feedback.

KernAttila commented:
> Found another thing when I clicked the scheduler menu item :
> ```python
> TypeError: CueNIMBYTray._toggle_scheduler() missing 2 required positional arguments: 'icon' and 'item'
> ```

Oops, remains from pystray 😅

KernAttila commented:
@lithorus @ramonfigueiredo here is the Launch CueGui option (will be disabled if CueGui is not found in PATH)
<img width="339" height="258" alt="image" src="https://github.com/user-attachments/assets/86613319-d224-49d9-b7c6-33f37aaf28f6" /> <img width="300" height="231" alt="image" src="https://github.com/user-attachments/assets/ea750522-d88d-4ae6-88ec-f55cdf894d1c" />



- I moved the config file to `~/.config/opencue/cuenimby.json`
- The labels are back to "Available" and "Notifications"

Also I fixed a little issue where the icon was not updating during initialization.

DiegoTavares reviewed (commented):
I'll leave for @ramonfigueiredo to approve this as he was involved in the initial development.

It would be nice to have some testing on a linux environment to confirm migrating from pystray to pyqt works as expected.

ramonfigueiredo commented:
Great work, @KernAttila.

There are still a few linting issues to fix.
Let’s close this PR for now, any additional changes can be submitted in a new one. Including https://github.com/AcademySoftwareFoundation/OpenCue/issues/2054, if you didn't start.

@KernAttila, please tag me here once you’ve finished and I will finish the code review and documentation. I’ll do the final review, update the code if needed, and merge it into the master branch.

Thanks!

lithorus commented:
> Hi @KernAttila,
> 
> Thanks for your excellent work. I'll perform the final code review now and update the documentation.
> 
> Initial tests on macOS and Linux are working well, everything looks good so far. I'll complete the final review and merge your code soon.
> 
> @lithorus @DiegoTavares 
> FYI 

I've also tested it on KDE and it's working good there too.

KernAttila commented:
@ramonfigueiredo 
Wow thanks for the full test, I'm glad it all works as expected :)
Also thanks for the nudge with pushing the first version of CueNimby recently <3

ramonfigueiredo commented:
Hi @KernAttila 

PR merged.

Thank you very much!

ramonfigueiredo commented on cuenimby/README.md:
```diff
@@ -213,6 +243,38 @@ This gives production teams fine-grained control over when and how much desktop
+| Icon | Name | Description |
+|------|------|-------------|
+| ![Available](cuenimby/icons/opencue-available.png) | `opencue-available.png` | Green icon - Host ready for rendering |
+| ![Working](cuenimby/icons/opencue-working.png) | `opencue-working.png` | Blue icon - Currently rendering |
```
Ok. I will submit a new PR with the change later

ramonfigueiredo commented:
New updates on CueNIMBY are posted on the OpenCue Slack.

https://academysoftwarefdn.slack.com/archives/CMFPXV39Q/p1761781601996439
